src/var.c

```diff
@@ -216,6 +217,14 @@ int DEFAULT_SYMVER_PRE(fi_param_define)(const struct fi_provider *provider,
 	v->name = strdup(param_name);
 	v->type = type;
 
+	entry = dlist_find_first_match(&param_list,param_match_name, v);
+
+        if (entry) {
+          FI_INFO(provider, FI_LOG_CORE,
+                "%s variable alredy exist\n",param_name);
+          fi_free_param(v);
+          return FI_SUCCESS;
```

Comment:
I would consider this an error on the part of the provider.  If we want to allow duplicate insertions, in order to return success, the rest of the fields should match., but at least the type.